### PR TITLE
Small bits of cleanup in sync'ing code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
I'd like to make some larger changes, but i'm breaking out small things i find along the way as i wrap my head around everything happening here.

### CODE REVIEW

File: src/VisualStudio/Core/Test.Next/Remote/SnapshotSerializationTests.cs [hunk 1]
```diff
@@ -83,14 +83,14 @@ public async Task CreateSolutionSnapshotId_Empty()
 
             using var scope = await validator.AssetStorage.StoreAssetsAsync(solution, CancellationToken.None).ConfigureAwait(false);
             var checksum = scope.SolutionChecksum;
-            var solutionSyncObject = await scope.GetAssetAsync(checksum, CancellationToken.None).ConfigureAwait(false);
+            var solutionSyncObject = await scope.GetTestAccessor().GetAssetAsync(checksum, CancellationToken.None).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Test.Next/Remote/SnapshotSerializationTests.cs:86]: moved functionality only used by tests to test helpers.

File: src/Workspaces/Remote/Core/SolutionAsset.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CodeAnalysis.Remote
     /// <summary>
     /// Represents a part of solution snapshot along with its checksum.
     /// </summary>
-    internal sealed class SolutionAsset
+    internal readonly struct SolutionAsset
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/SolutionAsset.cs:13]: no need for this tiny aggregate to cause allocations.

File: src/Workspaces/Remote/Core/SolutionAssetProvider.cs [hunk 3]
```diff
@@ -58,35 +58,16 @@ private async ValueTask WriteAssetsWorkerAsync(PipeWriter pipeWriter, Checksum s
...
             var scope = assetStorage.GetScope(solutionChecksum);
 
-            SolutionAsset? singleAsset = null;
-            PooledDictionary<Checksum, SolutionAsset>? assetMap = null;
+            using var _ = PooledDictionary<Checksum, SolutionAsset>.GetInstance(out var assetMap);
 
-            try
-            {
-
-                if (checksums.Length == 1)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/SolutionAssetProvider.cs:67]: no need to special case this.  we do all the same work effectively anways.

File: src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs [hunk 4]
```diff
@@ -48,65 +47,29 @@ internal sealed partial class Scope : IDisposable
         public void Dispose()
             => _storage.DecreaseScopeRefCount(this);
 
-        /// <summary>
-        /// Retrieve asset of a specified <paramref name="checksum"/> available within <see langword="this"/> from
-        /// the storage.
-        /// </summary>
-        public async ValueTask<SolutionAsset> GetAssetAsync(Checksum checksum, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs:55]: moved to test accessors.

File: src/Workspaces/Remote/Core/SolutionAssetStorage.cs [hunk 5]
```diff
@@ -115,18 +116,9 @@ internal TestAccessor(SolutionAssetStorage solutionAssetStorage)
             _solutionAssetStorage = solutionAssetStorage;
         }
 
-        public async ValueTask<SolutionAsset?> GetAssetAsync(Checksum checksum, CancellationToken cancellationToken)
+        public async ValueTask<SolutionAsset> GetRequiredAssetAsync(Checksum checksum, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/SolutionAssetStorage.cs:119]: this is inside a test accessor.  and all callers were throwing if we got null anyways.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**